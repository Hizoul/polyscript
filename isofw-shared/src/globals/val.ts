const val = {
  isDebug: true,
  version: "0.1.0",
  i18nNameSpace: "langSpace",
  channel: {
    realtime: "realtime"
  },
  service: {
    user: "/users",
    project: "projects",
    camera: "cameras",
    preset: "presets",
    presetAssistant: "presetAssistant"
  },
  maximumPresetAmount: 100
}

export default val
