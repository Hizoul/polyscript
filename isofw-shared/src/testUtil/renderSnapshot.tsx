import * as React from "react"
const render = require("preact-render-to-json")

const renderSnapshot = (component: any, msg: string) => {
  expect(render(component)).toMatchSnapshot(msg)
}

export default renderSnapshot
