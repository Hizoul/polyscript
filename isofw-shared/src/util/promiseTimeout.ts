const promiseTimeout = (waitTime: number) => {
  return new Promise((resolve) => {setTimeout(resolve, waitTime)})
}
export default promiseTimeout
