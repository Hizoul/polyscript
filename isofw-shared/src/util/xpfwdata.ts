import {
  addId, AuthForm, BackendClient, dataOptions, DbStore, displayModeChanger,
  getListFormFromRelationshipField, IBackendClient, ICreateHookProps, IEditHookProps,
  IGetHookProps, IListHookProps, IRelationshipHookProps, IRemoveHookProps, IUiClient,
  ListStore, MailField, OwnerField, PwField, REMOVE_ADDON_KEY, removeId, searchRelated,
  useAuth, useCreate, useCreateWithProps, useEdit, useEditWithProps, useGet, useGetWithProps, toJS,
  useList, useListWithProps, useRelationship, useRelationshipWithProps, useRemove, useRemoveWithProps, UserStore
} from "@xpfw/data"

export {
  addId, AuthForm, BackendClient, dataOptions, DbStore, displayModeChanger,
  getListFormFromRelationshipField, IBackendClient, ICreateHookProps, IEditHookProps,
  IGetHookProps, IListHookProps, IRelationshipHookProps, IRemoveHookProps, IUiClient,
  ListStore, MailField, OwnerField, PwField, REMOVE_ADDON_KEY, removeId, searchRelated,
  useAuth, useCreate, useCreateWithProps, useEdit, useEditWithProps, useGet, useGetWithProps, toJS,
  useList, useListWithProps, useRelationship, useRelationshipWithProps, useRemove, useRemoveWithProps, UserStore
}
