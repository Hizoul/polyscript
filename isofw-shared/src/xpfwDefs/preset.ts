import ValidationRegistry, { FieldType, IField, IForm, Permission } from "@xpfw/validate"
import val from "isofw-shared/src/globals/val"
import { cloneDeep } from "lodash"
import { IDField } from "./commonFields"

const EMPTY_PRESET = "-"

const PresetNumberField: IField = {
  type: FieldType.Number,
  mapTo: "number"
}

const PresetProjectField: IField = {
  type: FieldType.Text,
  mapTo: "project"
}

const PresetCameraField: IField = {
  type: FieldType.Text,
  mapTo: "camera"
}

const PresetForm: IForm = {
  model: "presetModel",
  collection: val.service.preset,
  sections: [{fields: [
    IDField, PresetNumberField, PresetProjectField, PresetCameraField
  ]}],
  permissions: {
    required: {
      create: Permission.User,
      find: Permission.User,
      get: Permission.User,
      update: Permission.User,
      remove: Permission.User
    }
  },
  options: {
    addCreatedAt: true,
    idPath: "_id",
    defaultSort: {
      [PresetCameraField.mapTo]: 1,
      [PresetNumberField.mapTo]: 1
    }
  }
}

ValidationRegistry.registerForm(PresetForm)

const PresetAssistantForm: IForm = cloneDeep(PresetForm)
PresetAssistantForm.collection = val.service.presetAssistant
ValidationRegistry.registerForm(PresetAssistantForm)

export {
  PresetForm, PresetNumberField, PresetProjectField, PresetCameraField, EMPTY_PRESET, PresetAssistantForm
}
