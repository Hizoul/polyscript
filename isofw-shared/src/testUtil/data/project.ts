import { MailField, PwField } from "@xpfw/ui-shared"
import val from "isofw-shared/src/globals/val"
import isServerParams from "isofw-shared/src/globals/isServerParams"
import { ProjectForm, ProjectName, ProjectShot } from "isofw-shared/src/xpfwDefs/project";

const testProjects = [
  {_id: "b19999999999999999999999", [ProjectName.mapTo]: "pn1", [ProjectShot.mapTo]: 1},
  {_id: "b29999999999999999999999", [ProjectName.mapTo]: "pn2", [ProjectShot.mapTo]: 8},
  {_id: "b39999999999999999999999", [ProjectName.mapTo]: "pn3", [ProjectShot.mapTo]: 6}
]
8
const createTestProjects = async (app: any) => {
  const newProjects = []
  for (const newUser of testProjects) {
    newProjects.push(await app.service(val.service.project).create(newUser, isServerParams))    
  }
  return newProjects
}

export default createTestProjects
export {
  testProjects
}