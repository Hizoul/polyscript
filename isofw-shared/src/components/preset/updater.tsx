import { DbStore } from "@xpfw/data"
import { memo, prependPrefix, useField } from "@xpfw/form"
import { cameraCommand } from "isofw-shared/src/cameraApi"
import val from "isofw-shared/src/globals/val"
import { PresetActionTypeField, PresetAssistantForm, PresetCameraField, PresetForm, PresetIsReadyField } from "isofw-shared/src/xpfwDefs/preset"
import { get } from "lodash"

const updatePrefix = "presetUpdater"
const presetFormHelper = useField(String(PresetForm.title), updatePrefix)
const presetAssistantFormHelper = useField(String(PresetAssistantForm.title), updatePrefix)
const presetIsReadyHelper = useField(String(PresetIsReadyField.title), prependPrefix(String(PresetForm.title), updatePrefix))

const updatePresetReadiness = (presetId: string, isReady: boolean) => {
  return async () => {
    presetFormHelper.setValue({isReady})
    const updateRes = await DbStore.patch(presetId, PresetForm, undefined, updatePrefix)
    return updateRes
  }
}
const savePresetData = (presetId: string) => {
  return async () => {
    presetAssistantFormHelper.setValue({
      [String(PresetActionTypeField.title)]: cameraCommand.updatePreset,
      [String(PresetCameraField.title)]: presetId
    })
    const preset = await DbStore.getFromServer(presetId, val.service.preset)
    const updateRes = await DbStore.patch(preset[String(PresetCameraField.title)], PresetAssistantForm, undefined, updatePrefix)
    return updateRes
  }
}

const usePresetUpdater = (id: string) => {
  const item = DbStore.getGetState(id, val.service.preset, true)
  return {
    item,
    isReady: get(item, String(PresetIsReadyField.title), false),
    setReady: memo(() => updatePresetReadiness(id, true), [updatePrefix, id, true]),
    setNotReady: memo(() => updatePresetReadiness(id, false), [updatePrefix, id, false]),
    savePreset: memo(() => savePresetData(id), ["savePreset", updatePrefix, id])
  }
}

export interface IPresetUpdaterProps {id: string}

const usePresetUpdaterWithProps = (props: IPresetUpdaterProps) => usePresetUpdater(props.id)

export default usePresetUpdater
export {
  updatePresetReadiness as updatePreset, usePresetUpdaterWithProps,
  savePresetData
}
