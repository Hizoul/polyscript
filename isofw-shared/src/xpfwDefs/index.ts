import { dataOptions } from "@xpfw/data"
dataOptions.idPath = "_id"

import "./camera"
import "./preset"
import "./project"
import "./user"
