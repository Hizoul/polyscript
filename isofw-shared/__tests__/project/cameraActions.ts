import { FeathersClient } from "@xpfw/data-feathers"
import { FormStore } from "@xpfw/form/dist"
import getRandomApp from "isofw-node/src/testUtil/getRandomApp"
import mockCameras from "isofw-node/src/testUtil/mockCameras"
import { savePresetData, updatePreset } from "isofw-shared/src/components/preset/updater"
import useCameraActions from "isofw-shared/src/components/project/cameraActions"
import val from "isofw-shared/src/globals/val"
import createTestCameras from "isofw-shared/src/testUtil/data/camera"
import createTestProjects, { testProjects } from "isofw-shared/src/testUtil/data/project"
import createTestUsers from "isofw-shared/src/testUtil/data/users"
import logIntoUser from "isofw-shared/src/testUtil/login"
import renderSnapshot from "isofw-shared/src/testUtil/renderSnapshot"
import promiseTimeout from "isofw-shared/src/util/promiseTimeout"
import { BackendClient, DbStore, ListStore } from "isofw-shared/src/util/xpfwdata"
import { PresetForm } from "isofw-shared/src/xpfwDefs/preset"
import { ProjectForm, ShotCamera } from "isofw-shared/src/xpfwDefs/project"
import * as MockDate from "mockdate"
import * as React from "react"

MockDate.set(new Date(4, 2, 0))

BackendClient.client = FeathersClient
const untypedDbStore: any = DbStore

describe("Camera actions", () => {
  it("works as planned", async () => {
    const appRef = await getRandomApp(" not important ", true, BackendClient.client, false)
    await createTestUsers(appRef.app)
    await logIntoUser()
    await createTestCameras(appRef.app)
    const mockedCameras = await mockCameras(appRef.app)
    const projectResults = await createTestProjects(appRef.app, true)
    const presetList = await ListStore.getList(PresetForm, undefined, undefined, true)
    await DbStore.getFromServer(testProjects[0]._id, val.service.project)
    FormStore.setValue(ProjectForm.title, projectResults[0])
    expect(mockedCameras[0].requests).toMatchSnapshot("requests before anything")
    const actionHelper = useCameraActions(ShotCamera, undefined, ProjectForm.title)
    await actionHelper.teleZoom()
    await actionHelper.wideZoom()
    await actionHelper.stopZoom()
    expect(mockedCameras[0].requests).toMatchSnapshot("requests After buttons")
    await promiseTimeout(1000)
    await appRef.cleanUp()
  }, 100000)
})
