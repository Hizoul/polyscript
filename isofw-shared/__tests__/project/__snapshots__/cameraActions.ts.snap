// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camera actions works as planned: requests After buttons 1`] = `
Array [
  Object {
    "params": Object {
      "0": "/cgi-bin/aw_ptz",
    },
    "query": Object {
      "cmd": "#Z45",
      "res": "1",
    },
  },
  Object {
    "params": Object {
      "0": "/cgi-bin/aw_ptz",
    },
    "query": Object {
      "cmd": "#Z55",
      "res": "1",
    },
  },
  Object {
    "params": Object {
      "0": "/cgi-bin/aw_ptz",
    },
    "query": Object {
      "cmd": "#Z50",
      "res": "1",
    },
  },
]
`;

exports[`Camera actions works as planned: requests before anything 1`] = `Array []`;
